custom-pages/universal-login/src/components/Register/Register.js

```diff
@@ -168,10 +168,17 @@ class Register extends Component {
     });
   };
 
-  goToAlert = (event) => {
-    if (event) event.preventDefault();
+  goToAlert = (errorName, event) => {
+    event.preventDefault();
 
-    getFirstErrorElement(this.state.errors).scrollIntoView({ block: 'center' })
+    document.getElementsByName(errorName)[0].scrollIntoView({ block: 'center' });
```

Comment:
document.getElementsByName seems wrong to use for a react app. I think this was used initially when we were targetting IE8. Now that we're targetting IE11+ it'd be nice to get rid of any accessing the DOM like that.

for scrolling, i think you should be using a React.createRef() - https://reactjs.org/docs/refs-and-the-dom.html

then you can do a window.scrollTo passing in the ref's offsettop.

what do you think about this idea? using the dom to get elements based off the name seems a bit non-react to me. I'm open to using it if it's the best way though.

Reply:
Yeah, I completely agree with this. I lifted the document.getElementsByName from the getFirstErrorElement function and not sure I would have approached it like that either. I will take a look at the above.